### PR TITLE
[CARBONDATA-571][CARBONDATA-572] Clean up code for carbon spark integration

### PR DESCRIPTION
**Pull request by jackylk**
1. remove unused declaration
2. remove redundant 'new' statement
3. fix generic compilation warning

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/387/

File: integration/spark-common/src/main/java/org/apache/carbondata/spark/load/DeleteLoadFolders.java [hunk 1]
```diff
@@ -60,11 +61,11 @@ private DeleteLoadFolders() {
    * returns segment path
    */
   private static String getSegmentPath(String dbName, String tableName, String storeLocation,
-      int partitionId, LoadMetadataDetails oneLoad) {
+      LoadMetadataDetails oneLoad) {
     CarbonTablePath carbon = new CarbonStorePath(storeLocation).getCarbonTablePath(
         new CarbonTableIdentifier(dbName, tableName, ""));
     String segmentId = oneLoad.getLoadName();
-    return carbon.getCarbonDataDirectoryPath("" + partitionId, segmentId);
+    return carbon.getCarbonDataDirectoryPath("" + 0, segmentId);
```
Comment (chenliang613) [inline integration/spark-common/src/main/java/org/apache/carbondata/spark/load/DeleteLoadFolders.java]: suggest keeping partitionId as input parameter.

Comment (jackylk) [inline integration/spark-common/src/main/java/org/apache/carbondata/spark/load/DeleteLoadFolders.java]: fixed

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/427/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**